lib/voyager/clients/microsoft_graph_client.rb

```diff
@@ -39,52 +39,53 @@ def account_info(*fields)
       get('/me', select: fields)
     end
 
+    def search(options = {})
+      post('/search/query', options[:body], options[:headers] || {})
+    end
+
     # ============================================================================
     # Files
     # ============================================================================
 
-    def drives
+    def my_drives
       get('/me/drives')
     end
 
-    def drive
+    def my_drive
       get('/me/drive')
     end
 
-    def drive_items(item_id = 'root', options = {})
-      get("/me/drive/#{item_id}/children", options)
+    def my_drive_items(item_id = 'root', options = '')
```

Comment:
We should opt for passing the query params as a hash rather than a String and then convert within the method to the relevant query strings. The hash interface is cleaner than a String interface.

`client.my_drive_items('root', expand: 'children', token: 'blah', hi: 'there')` vs `client.my_drive_items('root', '?expand=children&token=blah&hi=there`.

Reply:
This makes it less user-friendly by hard-coding `expand` in there, when we might well want to change that to `select` or nest the expands & selects.

Reply:
You don't have to hardcode anything, you simply allow a Hash with keys that will get converted into a querystring. You can accept any key you want such as `select` or `expand` or `filter`. It's similar to requiring a string for activerecord where method. `User.where("name = 'Artin' AND knowledge_level = 'None')` vs `User.where(name: 'Artin', knowledge_level: 'None').

Reply:
I think I see what you are talking about. I see this example in the API Docs about nested select within filter:

`GET https://graph.microsoft.com/v1.0/me/drive/root?$expand=children($select=id,name)`

I think it's fine to have `client.search(expand: 'children($select=id,name)')`. I'm sure we can do fancier things but I don't think it makes sense to do it in this ticket like `client.search(expand: { children: { select: %w[id name] }})`. But if you decide to stick with a String that's fine with me.